core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PushJoinIntoTableScan.java

```diff
@@ -114,6 +115,19 @@ public Result apply(JoinNode joinNode, Captures captures, Context context)
             return Result.empty();
         }
 
+        if (getJoinPushdownMode(context.getSession()) == JoinPushdownMode.AUTOMATIC) {
+            PlanNodeStatsEstimate joinStats = context.getStatsProvider().getStats(joinNode);
+            PlanNodeStatsEstimate leftStats = context.getStatsProvider().getStats(joinNode.getLeft());
+            PlanNodeStatsEstimate rightStats = context.getStatsProvider().getStats(joinNode.getRight());
+            if (joinStats.isOutputRowCountUnknown() || leftStats.isOutputRowCountUnknown() || rightStats.isOutputRowCountUnknown()) {
```

Comment:
Since stats calculation can be costly (eg can involve a trip to metastore), short-circuit calculation as early as you can.
To keep this readable, please extract the condition to a separate method.